utilities/src/main/java/io/syndesis/qe/endpoints/publicendpoint/EnvironmentsPublicEndpoint.java

```diff
@@ -0,0 +1,65 @@
+package io.syndesis.qe.endpoints.publicendpoint;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.syndesis.common.util.Json;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.List;
+
+import static org.junit.Assert.fail;
+
+@Component
+@Lazy
+public class EnvironmentsPublicEndpoint extends PublicEndpoint {
+
+    public EnvironmentsPublicEndpoint() {
+        super("/environments");
+    }
+
+    /**
+     * Get all tags in the syndesis
+     * endpoint -> GET ​/public​/environments
+     * original method -> {@link io.syndesis.server.endpoint.v1.handler.external.PublicApiHandler#getReleaseEnvironments()}
+     */
+    public List<String> getAllEnvironments() throws IOException {
+        Invocation.Builder invocation = this.createInvocation(getWholeUrl(rootEndPoint));
+        Response response = invocation.get();
+        if (response.getStatus() != HttpURLConnection.HTTP_OK) {
+            fail("Status of the response is " + response.getStatus() + ". The response is " + response.getStatusInfo());
+        }
+        return Json.reader().forType(List.class).readValue(response.readEntity(JsonNode.class));
+    }
+
+    /**
+     * Delete particular tag. The tag is deleted from all integration where was tagged.
+     * endpoint -> DELETE /public​/environments​/{env}
+     * original method -> {@link io.syndesis.server.endpoint.v1.handler.external.PublicApiHandler#deleteEnvironment(String)}
+     */
+    public void deleteEnvironment(String environment) {
+        Invocation.Builder invocation = this.createInvocation(getWholeUrl(String.format(rootEndPoint + "/%s", environment)));
+        Response response = invocation.delete();
+        if (response.getStatus() != HttpURLConnection.HTTP_NO_CONTENT) {
+            fail("Status of the response is " + response.getStatus() + ". The response is " + response);
+        }
+    }
+
+    /**
+     * Rename particular tag.
+     * endpoint -> PUT /public​/environments​/{env}
+     * original method -> {@link io.syndesis.server.endpoint.v1.handler.external.PublicApiHandler#renameEnvironment(String, String)}
+     */
+    public void renameEnvironment(String environment, String newEnvironment) {
+        Invocation.Builder invocation = this.createInvocation(getWholeUrl(String.format(rootEndPoint + "/%s", environment)));
+        Response response = invocation.put(Entity.entity(newEnvironment, MediaType.APPLICATION_JSON));
+        if (response.getStatus() != HttpURLConnection.HTTP_NO_CONTENT) {  //404 #gh-
```

Comment:
there is something missing in the comment :yum:

Reply:
It shouldn't be there :(